src/tools/allocator-recipe-resolver.ts

```diff
@@ -209,3 +207,16 @@ export function findLongRunningArcId(recipe: Recipe): string | null {
   const arcIdAnnotation = recipe.getAnnotation('arcId');
   return arcIdAnnotation ? Object.values(arcIdAnnotation.params)[0].toString() : null;
 }
+
+/** Intermediary in the recipe resolution process that holds stores. */
+class ManifestStoreContext implements StoreContext {
+  constructor(readonly context: Manifest) {}
+
+  findStoreById(id: string): AbstractStore {
+    return this.context.findStoreById(id);
```

Comment:
Is this correct? In the light of my outline of 2 parts of the context in the comment above I think this may be wrong.

Reply:
I produced this by reading what `findStoreById` in an Arc was doing, omitting the irrelevant logic.

Reply:
This matches the current logic but I think it's actually a bug, in that `use` should never end up referring to anything that isn't actually in the current arc. If it works in our tests for `ManifestContext` to return an empty list here then do that in preference. In future refactors we can firm up this interface and allow solid specification of whether you want to fetch a local, non-local, or don't-care store.

Reply:
👍